addon/types/map_export/brush.py

```diff
@@ -40,7 +40,21 @@ def sort_into_parts(bm: bmesh.types.BMesh):
     return parts
 
 
-def calc_uv_axes(settings: typing.Any, bm: bmesh.types.BMesh, face: bmesh.types.BMFace):
+def get_texture_size(obj: bpy.types.Object, face: bmesh.types.BMFace):
+    texture_side_length = -1
+
+    if face.material_index < len(obj.material_slots):
+        face_material = obj.material_slots[face.material_index].material
+        if face_material and face_material.use_nodes:
+            for mat_node in face_material.node_tree.nodes:
+                if mat_node.type == 'TEX_IMAGE':
+                    texture_side_length = mat_node.image.size[0]
```

Comment:
I'm curious why the width and not the height, but you don't have to explain if you don't feel like it.